### ISSUE TITLE
Dependencies lost when publishing from PSDrive

### ISSUE THREAD

**Issue by adrian-andersson** [aws/aws-lambda-dotnet#391]
Whats Happening:

When publishing module via a PSDrive path instead of a direct path, module dependencies are not correctly compiled into the ZIP file.

Suspected Cause:
File : _DeploymentFunctions.ps1
Line: 436
Command: $ast = [System.Management.Automation.Language.Parser]::ParseFile($Script, [ref]$null, [ref]$null)

Issue:
Parameter $script is sending relative path.

Possible solution:
Could be fixed by resolving the full path of $script, e.g. $script = $(get-item $script).fullname
This should happen in the base function param and be passed to all child functions

Workaround:
Specify full path of $script before passing to LAMBDA

Concern:
Function completes successfully without errors. No indication that LAMBDA will not work until executed

Replicate: 
Create a lambda from an indirect path or PSDrive

In other words, its working if I use a proper path (I.e. c:\scripts\), not from a PS drive (I.e. github:\scripts)

Possibly related Issue:
https://github.com/aws/aws-lambda-dotnet/issues/349
^ Compiling from a network drive ^

**Comment by mtboren** [comment]
I think you nailed it, @adrian-andersson -- in function `_prepareDependentPowerShellModules` (in ` _DeploymentFunctions.ps1`), when `$Script` is the PowerShell-specific path (as returned by the `Resolve-Path` call in the module) to some file that resides at a UNC path or on a PSDrive, that invocation of ` [System.Management.Automation.Language.Parser]::ParseFile( )` is getting a path that .NET does not understand.

I'm working on resolution in a fork.  We'll see if that gets merged in via PR.
